packages/loader/container-loader/src/container.ts

```diff
@@ -387,7 +387,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
     private readonly mc: MonitoringContext;
 
-    private _lifecycleState: "loading" | "loaded" | "closing" | "closed" | "disposed" = "loading";
+    private _lifecycleState: "loading" | "loaded" | "closing" | "closed" | "disposing" | "disposed" = "loading";
```

Comment:
Can you please add documentation here on the differences between all these modes?
At least the intention - where do we expect differences at Container level between "disposing" and "closing"?
Ideally, we also document all allowed state transitions (i.e. there is no way to get from "closed" to "loading").

When the number of states is small and they are very distinct, we can get away with relying on next developer doing code inspection to figure out intent. But I feel like we are somewhat beyond that with what I see here.

Reply:
And more on point - I do not see how code handles differently "disposing" vs. "closing". If there are no differences, can we collapse these states into one?